### PR TITLE
Add an Endpoints Proto file

### PR DESCRIPTION
**Pull request by tgroh**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
This contains the APIServiceDescriptor proto, which is used for
specifying an endpoint to communicate to.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles @lukecwik 

This contains no actual population of the payload field or the Endpoints.ApiServiceDescriptor proto, nor any use of them. That will come in a follow-up, once a name and everything is settled on.

File: sdks/common/fn-api/src/main/proto/beam_fn_api.proto [hunk 1]
```diff
@@ -709,15 +710,19 @@ message ApiServiceDescriptor {
   // refer to this.
   string id = 1;
 
-  // (Required) The URL to connect to.
-  string url = 2;
+  // (Required) The URL to connect to. Deprecated: use payload instead
+  string url = 2 [deprecated = true];
```
Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: I think we want to replace the ApiServiceDescriptor here with the one you defined so just mark this entire message as deprecated saying to migrate to the definition in `endpoints.proto`. The `id` field is not actually used anywhere meaningfully.

**Comment by tgroh** [comment]
Done.

**Comment by tgroh** [comment]
Fixed python also

**Comment by lukecwik** [comment]
The proto change is reasonable to me, I'm assuming that either Ken or you will resolve the changes needed for pr/3924

**Comment by tgroh** [comment]
retest this please

**Comment by tgroh** [comment]
retest this please

**Comment by tgroh** [comment]
retest this please

**Pull request closed without merging**